src/Logic.hs

```diff
@@ -488,10 +489,32 @@ proceedUntilFixedPoint state = do
     then return state
     else proceedUntilFixedPoint newState
 
+-- Provide feedback for pull requests where 'needsFeedback' is set.
+feedback :: (PullRequestId, PullRequest) -> ProjectState -> Action ProjectState
+feedback (prId, pr) state = case Pr.classifyPullRequest pr of
```

Comment:
I think we could split this into two parts:

* A pure function `describeStatus :: (PullRequestId, PullRequest) -> ProjectState -> Text`
* A function that walks all pull requests that need feeback, leaves a comment with the result of `describeStatus`, and then unsets the flag.

`describeStatus` should exhaustively describe every status that a pull request can be in, in particular the situation where a pull request is approved and integrated, but the build failed. That way, when you leave an approval comment on such a PR, Hoff would reply that it’s not going to take any further action on the pull request.

It might even be possible to convert all existing cases where we leave a comment into this feedback mechanism.